app/models/katello/glue/pulp/repo.rb

```diff
@@ -215,11 +215,11 @@ def importer_ssl_options(capsule = SmartProxy.default_capsule!)
             :ssl_client_key => ueber_cert[:key],
             :ssl_ca_cert => ::Cert::Certs.ca_cert
           }
-        elsif self.try(:redhat?) && self.content_view.default?
+        elsif self.try(:redhat?) && self.content_view.default? && Katello::Resources::CDN::CdnResource.redhat_cdn?(url)
           importer_options = {
             :ssl_client_cert => self.product.certificate,
             :ssl_client_key => self.product.key,
-            :ssl_ca_cert => Resources::CDN::CdnResource.ca_file_contents
+            :ssl_ca_cert => Katello::Repository.feed_ca_cert(url)
           }
         else
```

Comment:
A better phrasing of my question --> It seems like the block below assumes http, is that still correct with your additions?